src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -333,6 +345,28 @@ private void OnDataReceived(object sender, ProcessDataReceivedEventArgs e)
             }
         }
 
+        private bool isNode20SupportedSystems() {
+            var systemName = PlatformUtil.GetSystemId();
+            var systemVersion = PlatformUtil.GetSystemVersion().Name.ToString();
+            if (systemName.Equals("ubuntu") &&
+                int.TryParse(systemVersion, out int ubuntuVersion) &&
+                ubuntuVersion <= 18.04) {
+                Trace.Info($"Detected Ubuntu version <= 18.04, using node16 as execution handler, instead node20");
+                return false;
+            }
+            if (systemName.Equals("debian") &&
+                int.TryParse(systemVersion, out int debianVersion) &&
+                debianVersion <= 9) {
+                Trace.Info($"Detected Debian version <= 9, using node16 as execution handler, instead node20");
+                return false;
+            } 
+            if (PlatformUtil.RunningOnRHELVersion("6") || PlatformUtil.RunningOnRHELVersion("7")) {
```

Comment:
>("6" [](http://example.com/codeflow?start=49&length=4)

should be "6." or "7." to match existing logic in platformutil

Reply:
updated